src/Voter/TimesheetVoter.php

```diff
@@ -173,4 +191,18 @@ protected function canDelete(User $user, Timesheet $timesheet): bool
 
         return true;
     }
+
+    protected function canDuplicate(User $user, Timesheet $timesheet): bool
+    {
+        if ($timesheet->isExported() && !$this->hasRolePermission($user, 'edit_exported_timesheet')) {
```

Comment:
Lets re-use the code:
```suggestion
        if (!$this->canEdit($user, $timesheet)) {
```
but actually ... hm ... I don't think that the check is even required.
Duplicating a timesheet shouldn't be a problem, even if the original one is exported. Or do you see a good reason for preventing it?

Reply:
In my opinion duplicating is a problem. Because the time get's doubled. Let's assume punch mode is active. Users can only punch in and out.

If they punch in and out they are only able to creat times when they punch in -> out.

Looks on a normal day like this:
![grafik](https://user-images.githubusercontent.com/36242595/95472725-aba4c480-0983-11eb-8c05-a09acfbc420b.png)

if they duplicate an entry - the time get's doubled. So if you check at the end of the month you just can't check the total duration if it matches daily working time * working days of current month -> you have to check for overlapping times, too.

-----

Just found out copying is not possible if "allow overlapping time entries" is disabled (in latest master) but possible in latest stable. Guess this is related to the same thing we already discussed last week or so.

How to proceed? Close pull request and delete it?

Reply:
I wasn't questioning this PR.
What I meant is: do we need the check about the export state in the voter. 

I think it is valid to hide the button in punch-in-out mode.